packages/workspace/src/expressions.ts

```diff
@@ -131,14 +131,15 @@ const resolveElement = (
   if (isVariable(element)) {
     element.value = _.cloneWith(element.value, referenceCloner)
   }
-
+  element.annotationTypes = _.cloneDeep(element.annotationTypes)
```

Comment:
we should never run `cloneDeep` on elements as it "disconnects" the references (creates a million "builtin string" types).
there is another fix for a similar issue to the one being created here (where we disconnect the field from its proper parent because of the cloneDeep) but unfortunately it is stuck in [this PR](https://github.com/salto-io/salto/pull/1215) and mixed in with a ton of other stuff...

we can try to leave that bug in place and only resolve the issue with the types
```suggestion
  element.annotationTypes = _.mapValues(element.annotationTypes, type => contextElements[type.elemID.getFullName()])
```

but I suggest push this whole bug fix in a new PR